build.js

```diff
@@ -0,0 +1,12 @@
+const childProcess = require("child_process");
+
+const env = {
+  ...process.env,
+  // Avoid embedding timestamps into linked binary on macOS
+  ZERO_AR_DATE: "1",
```

Comment:
Can we set this as an environment variable directly in `bindings.gyp`? I would rather not involve another script if it just changes the environment like this.

Also, is this something we want for _all_ builds, or is reproducible builds something for specific scenarios?

Reply:
If you find a way to do it in a cross-platform way, I'd be happy to change it! However, setting env variables on Linux/macOS is with `ZERO_AR_DATE=1`, and on Windows it's `$env:ZERO_AR_DATE=1` (at least in PowerShell, I'm not sure about CMD).

> Also, is this something we want for all builds, or is reproducible builds something for specific scenarios?

I don't see any benefit in having timestamps embedded in the binaries. As such, I'd always set this variable (unless there's a scenario where embedded timestamps would be needed).

Reply:
Would it work if it was set to the GitHub [workflow step `environment`](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#jobsjob_idenvironment)? That would be available on build when released, but local builds will still contain timestamps unless the user also sets `ZERO_AR_DATE=1`

Alternatively, I believe that npm scripts are portable and setting the build script to

```
    "install": "ZERO_AR_DATE=1 node-gyp-build",
```

works (at least it works with yarn)

(sorry for the late reply)

Reply:
TIL about cross-env, that package looks nice! Yes, feel free to update my branch as desired 🙂